### PR TITLE
Added roles. Changed login behavior based on user role. Added passwor…

### PR DESCRIPTION
**Pull request by Ravennod**
…ds in seed method.

### CODE REVIEW

File: MedicalDocManagment.WebUI/Providers/AuthorizationServerProvider.cs [hunk 1]
```diff
@@ -12,27 +16,46 @@ public class AuthorizationServerProvider : OAuthAuthorizationServerProvider
         {
             context.Validated();
         }
+        public override Task TokenEndpoint(OAuthTokenEndpointContext context)
+        {
+            foreach (KeyValuePair<string, string> pair in context.Properties.Dictionary)
```
Comment (artempanko) [inline MedicalDocManagment.WebUI/Providers/AuthorizationServerProvider.cs]: var

File: MedicalDocManagment.WebUI/Providers/AuthorizationServerProvider.cs [hunk 2]
```diff
@@ -12,27 +16,46 @@ public class AuthorizationServerProvider : OAuthAuthorizationServerProvider
...
             if (user == null)
             {
                 context.SetError("invalid_grant", "The user name or password is incorrect.");
                 return;
             }
 
-            var identity = new ClaimsIdentity(context.Options.AuthenticationType);
-            identity.AddClaim(new Claim("sub", context.UserName));
-            identity.AddClaim(new Claim("role", "user"));
+            List<Claim> roles = oAuthIdentity.Claims.Where(c => c.Type == ClaimTypes.Role).ToList();
```
Comment (artempanko) [inline MedicalDocManagment.WebUI/Providers/AuthorizationServerProvider.cs]: var

File: MedicalDocManagment.WebUI/Providers/AuthorizationServerProvider.cs [hunk 3]
```diff
@@ -12,27 +16,46 @@ public class AuthorizationServerProvider : OAuthAuthorizationServerProvider
...
+            AuthenticationProperties properties = CreateProperties(user.UserName, Newtonsoft.Json.JsonConvert.SerializeObject(roles.Select(x => x.Value)));
 
-            context.Validated(identity);
+            AuthenticationTicket ticket = new AuthenticationTicket(oAuthIdentity, properties);
+            context.Validated(ticket);
 
         }
+        public static AuthenticationProperties CreateProperties(string userName, string Roles)
+        {
+            IDictionary<string, string> data = new Dictionary<string, string>
```
Comment (artempanko) [inline MedicalDocManagment.WebUI/Providers/AuthorizationServerProvider.cs]: var

File: MedicalDocManagment.WebUI/app/core/login/authentication.service.ts [hunk 4]
```diff
@@ -26,12 +29,14 @@ export class AuthenticationService {
                             console.log(response);
                             // login successful if there's a jwt token in the response
                             let token = response.json() && response.json().access_token;
+                            this.roles = new Array(response.json().roles.replace(/[\"\[\]]+/g, '').split(","));
```
Comment (artempanko) [inline MedicalDocManagment.WebUI/app/core/login/authentication.service.ts]: create method

**Review by artempanko — APPROVED** [review]

**Review by KostetskyiYaroslav — APPROVED** [review]

**Pull request merged**